pagopa/io-app
# chore: [IOAPPCOM-19] Due date banner on prescription message detail is not aligned with the one on other messages' detail.

## Short description
This PR aligns the due date banner `MedicalPrescriptionDueDateBar.tsx` on the prescription message detail to the one present on the other messages' detail screen (Rif. #4190)

| &nbsp; | Android | iOS |
| - | - | - |
| Valid/Expiring | <img src="https://user-images.githubusercontent.com/16268789/207334190-6d80a538-89c3-4634-8e6a-6dc37e2d4018.png" width="150" /> | <img src="https://user-images.githubusercontent.com/16268789/207334211-a7f50085-c0f4-4a84-8291-c3d2d9f85108.png" width="150" />
| Expired | <img src="https://user-images.githubusercontent.com/16268789/207334172-7a1328aa-b94f-4744-bb3e-1afa1418f975.png" width="150" /> | <img src="https://user-images.githubusercontent.com/16268789/207334198-63a93199-a113-433e-ab32-7c999ca53432.png" width="150" />

## List of changes proposed in this pull request

-  `locales`
    - `it/index.yml`: updated locale strings as per #4190.
    - `en/index.yml`: updated locale strings as per #4190.
    - `de/index.yml`: removed no more used locale strings.
- `ts/components/messages/MedicalPrescriptionDueDateBar.tsx`: updated with the new locale labels, even if this is a _to-delete_ code, just to make it compile.
- `ts/components/messages/paginated/MedicalPrescriptionDueDateBar.tsx`: updated the banner layout to the one made on #4190, using the new/updated locale strings.

## How to test
Using [dev-io-api-server](https://github.com/pagopa/io-dev-api-server) add prescription messages with different due dates and payment information with `invalidAfterDueDate` set to `true`, like below:

```ts
// Valid
const medicalMessage = (count: number) =>
  withDueDate(
    withPaymentData(
      getNewMessage(
        customConfig,
        `💊 medical prescription - ${count}`,
        messageMarkdown,
        medicalPrescription
      ),
      true // invalidAfterDueDate
    ),
    new Date(now.getTime() + 60 * 1000 * 60 * 24 * 8) // dueDate 
);
// Expired
const medicalMessage = (count: number) =>
  withDueDate(
    withPaymentData(
      getNewMessage(
        customConfig,
        `💊 medical prescription - ${count}`,
        messageMarkdown,
        medicalPrescription
      ),
      true // invalidAfterDueDate
    ),
    new Date(now.getTime() - 60 * 1000 * 60 * 24 * 8) // dueDate 
);
```

and check on both Android and iOS that the prescription message detail screen is rendered like the screenshots above.
